plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/commands/SQLCommandInclude.java

```diff
@@ -77,76 +72,113 @@ public SQLControlResult handleCommand(@NotNull DBRProgressMonitor monitor, @NotN
-                            }
-                        }
-                    }
-                }
-                final IncludeEditorInput input = new IncludeEditorInput(finalIncFile, fileContents);
-                SQLEditor sqlEditor = SQLEditorHandlerOpenEditor.openSQLConsole(
-                        workbenchWindow,
-                        new SQLNavigatorContext(scriptContext, true),
-                        input);
-                sqlEditor.reloadSyntaxRules();
+                closeDuplicatedEditors(finalIncFile);
+                SQLEditor sqlEditor = getSqlEditor(scriptContext, finalIncFile, workbenchWindow);
                 final IncludeScriptListener scriptListener = new IncludeScriptListener(
                     workbenchWindow,
                     sqlEditor,
-                    statusFlag);
+                    result
+                );
                 boolean execResult = sqlEditor.processSQL(false, true, null, scriptListener);
                 if (!execResult) {
-                    statusFlag[0] = true;
+                    result.complete(SQLControlResult.failure());
                 }
             } catch (Throwable e) {
                 log.error(e);
-                statusFlag[0] = true;
+                result.complete(SQLControlResult.failure());
             }
         });
+        return result;
+    }
 
-        // Wait until script finished
-        while (!statusFlag[0]) {
-            try {
-                Thread.sleep(50);
-            } catch (InterruptedException e) {
-                break;
+    @NotNull
+    private SQLEditor getSqlEditor(
+        @NotNull SQLScriptContext scriptContext,
+        @NotNull Path finalIncFile,
+        @NotNull IWorkbenchWindow workbenchWindow
+    ) {
+        IncludedScriptFileEditorInput input = new IncludedScriptFileEditorInput(finalIncFile);
+        SQLEditor sqlEditor = SQLEditorHandlerOpenEditor.openNewSQLConsole(
+            workbenchWindow,
+            new SQLNavigatorContext(scriptContext, true),
+            input
+        );
+        sqlEditor.reloadSyntaxRules();
+        return sqlEditor;
+    }
+
+    private void closeDuplicatedEditors(@NotNull Path finalIncFile) throws PartInitException {
+        for (IWorkbenchWindow window : PlatformUI.getWorkbench().getWorkbenchWindows()) {
+            for (IWorkbenchPage page : window.getPages()) {
+                for (IEditorReference editorReference : page.getEditorReferences()) {
+                    if (isEditorForSameIncludedScript(editorReference, finalIncFile)) {
+                            UIUtils.syncExec(
+                                () -> page.closeEditor(editorReference.getEditor(false), false));
+
+                    }
+                }
             }
         }
+    }
 
-        return SQLControlResult.success();
+    private boolean isEditorForSameIncludedScript(@NotNull IEditorReference editorReference, @NotNull Path finalIncFile)
+    throws PartInitException {
```

Comment:
Parameter naming as before, plus code style
```suggestion
    private boolean isEditorForSameIncludedScript(
        @NotNull IEditorReference editorReference,
        @NotNull Path finalIncFile
    ) throws PartInitException {
```